BDArmory/Guidances/MissileGuidance.cs

```diff
@@ -147,6 +147,184 @@ public static Vector3 GetAirToGroundTarget(Vector3 targetPosition, Vector3 targe
+                            Vector3 accVec = accel*(Vector3.Dot(newVel, planarDirectionToTarget) * upDirection - Vector3.Dot(newVel, upDirection) * planarDirectionToTarget).normalized;
+
+                            return missileVessel.transform.position + 1.5f * Time.fixedDeltaTime * newVel + 2.25f * Time.fixedDeltaTime * Time.fixedDeltaTime * accVec;
+                            */
+                        /*}
+                        return missileVessel.transform.position + 0.5f * (float)missileVessel.srfSpeed * ((Mathf.Cos(loftAngle * Mathf.Deg2Rad) * planarDirectionToTarget) + (Mathf.Sin(loftAngle * Mathf.Deg2Rad) * upDirection));
+                        */
+                    //}
+
+                    //Debug.Log("[BDArmory.MissileGuidance]: AGM altitudeClamp =" + altitudeClamp);
+                    //Vector3 finalTarget = missileVessel.transform.position + 0.5f * (float)missileVessel.srfSpeed * planarDirectionToTarget + ((altitudeClamp - (float)missileVessel.altitude) * upDirection.normalized);
+
+                    //return finalTarget;
+                }
+                else
+                {
+                    loftState = 2;
+
+                    // Tried to do some kind of pro-nav method. Didn't work well, leaving it just in case I want to fix it.
+                    /*
+                    Vector3 newVel = (float)missileVessel.srfSpeed * velDirection;
+                    Vector3 accVec = (newVel - missileVessel.Velocity());
+                    Vector3 unitVel = missileVessel.Velocity().normalized;
+                    accVec = accVec - unitVel * Vector3.Dot(unitVel, accVec);
+
+                    float accelTime = Mathf.Clamp(timeToImpact, 0f, 4f);
+
+                    accVec = accVec / accelTime;
+
+                    float accel = accVec.magnitude;
+
+                    if (accel > 20f * (float)PhysicsGlobals.GravitationalAcceleration)
+                    {
+                        accel = 20f * (float)PhysicsGlobals.GravitationalAcceleration / accel;
+                    }
+                    else
+                    {
+                        accel = 1f;
+                    }
+
+                    Debug.Log("[BDArmory.MissileGuidance]: Loft: Diving, accel = " + accel);
+                    return missileVessel.transform.position + 1.5f * Time.fixedDeltaTime * missileVessel.Velocity() + 2.25f * Time.fixedDeltaTime * Time.fixedDeltaTime * accVec * accel;
+                    */
+                    return 0.5f * (float)missileVessel.srfSpeed * velDirection;
+                }
+            }
+            else
+            {
+                // If terminal just go straight for target + lead
+                loftState = 3;
+                if (BDArmorySettings.DEBUG_MISSILES) Debug.Log("[BDArmory.MissileLauncher]: Terminal");
+                return targetPosition + targetVelocity * leadTime + 0.5f * leadTime * leadTime * targetAcceleration;
```

Comment:
`AIUtils.PredictPosition(targetPosition, targetVelocity, targetAcceleration, leadTime);` does this calculation and is easier to read.

Reply:
Fixed